src/ui/customBreakpointsUI.ts

```diff
@@ -93,34 +164,92 @@ export function registerCustomBreakpointsUI(
 ) {
   const provider = new BreakpointsDataProvider(debugSessionTracker);
 
-  vscode.window.createTreeView(CustomViews.BrowserBreakpoints, {
+  const view = vscode.window.createTreeView(CustomViews.EventListenerBreakpoints, {
     treeDataProvider: provider,
+    showCollapseAll: true,
+    manageCheckboxStateManually: true,
   });
+
+  view.onDidChangeCheckboxState(e => {
+    if (e.items.length == 1 && e.items[0][0] instanceof BreakpointCategory) {
+      if (e.items[0][1]) {
+        provider.addBreakpoints(e.items[0][0].children);
+      } else {
+        provider.removeBreakpoints(e.items[0][0].children);
+      }
+      return;
+    }
+
+    for (const item of e.items) {
+      if (item[0] instanceof Breakpoint) {
+        if (item[1]) {
+          provider.addBreakpoints([item[0]]);
+        } else {
+          provider.removeBreakpoints([item[0]]);
+        }
+      }
+    }
+  });
+
+  context.subscriptions.push(view);
+
   context.subscriptions.push(
     vscode.commands.registerCommand(Commands.AddCustomBreakpoints, () => {
-      const quickPick = vscode.window.createQuickPick();
-      const items = provider.breakpoints.filter(b => !b.enabled);
-      quickPick.items = items;
-      quickPick.onDidAccept(() => {
-        provider.addBreakpoints(quickPick.selectedItems as Breakpoint[]);
-        quickPick.dispose();
-      });
-      quickPick.show();
+      const items = provider.breakpointCategories
+        .sort(BreakpointCategory.compare)
```

Comment:
This mutates the `breakpointCategories` on the provider, which we should avoid. (If you move to a map you get this automatically since this becomes `[...provider.breakpointCategories.values()]`)

You could also use the `QuckPickItemKind.Separator` to retain information about the groups 🙂

Reply:
included in [this commit](https://github.com/microsoft/vscode-js-debug/pull/1853/commits/f66568d03d1fe72d63d4141ffa3f3a8b034206bd)